api/v1alpha1/gateway_types.go

```diff
@@ -238,6 +238,39 @@ type LocalObjectReference struct {
 	Name string `json:"name" protobuf:"bytes,3,opt,name=name"`
 }
 
+// RouteBindingSelector defines a schema for associating routes with the Gateway.
+// No routes will be associated to the Gateway if NamespaceSelector and RouteSelector
+// are undefined. If both NamespaceSelector and RouteSelector are defined, only
+// routes matching the RouteSelector within the NamespaceSelector namespaces are
+// associated with the Gateway.
+type RouteBindingSelector struct {
+	// NamespaceSelector specifies a set of namespace labels used for selecting
+	// routes to associate with the Gateway. If NamespaceSelector is defined,
+	// all routes within the NamespaceSelector namespaces are associated with
+	// the Gateway. An empty NamespaceSelector matches all namespaces.
+	//
+	// If undefined, namespace labels are not used for associating routes to
+	// the gateway.
+	//
+	// Support: Core
+	//
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector" protobuf:"bytes,1,opt,name=namespaceSelector"`
+	// RouteSelector specifies a set of route labels used for selecting
+	// routes to associate with the Gateway. If RouteSelector is defined,
+	// only routes matching the RouteSelector are associated with the Gateway.
+	// An empty RouteSelector matches all routes.
+	//
+	//
+	// If undefined, route labels are not used for associating routes to
+	// the gateway.
+	//
+	// Support: Core
+	//
+	// +optional
+	RouteSelector *metav1.LabelSelector `json:"routeSelector" protobuf:"bytes,2,opt,name=routeSelector"`
```

Comment:
how do we handle priority between route? or routes is defined to be independent?
e.g. /path route and /* route have priority requirements to work correctly

Reply:
`RouteSelector` is a selector for associating routes to a gateway. The following example associates route "foo" to gateway "my-gateway" but not route "bar":
```
kind: Gateway
apiVersion: networking.x.k8s.io/v1alpha1
metadata:
  name: my-gateway
  namespace: default
spec:
  class: acme-lb
  listeners:  # Use GatewayClass defaults for listener definition.
  - name: my-http-listener
    protocol: HTTP
  routes:
    routeSelector:
      matchLabels:
        app: foo
---
kind: HTTPRoute
apiVersion: networking.x.k8s.io/v1alpha1
metadata:
  name: foo
  namespace: default
  labels:
    app: foo
---
kind: HTTPRoute
apiVersion: networking.x.k8s.io/v1alpha1
metadata:
  name: bar
  namespace: default
  labels:
    app: bar
```

Reply:
yes, is there an issue already regarding conflict resolution, should link it here.